library/pkcs12.c

```diff
@@ -212,13 +232,15 @@ int mbedtls_pkcs12_pbe( mbedtls_asn1_buf *pbe_params, int mode,
         goto exit;
 
     if( ( ret = mbedtls_cipher_update( &cipher_ctx, data, len,
-                                output, &olen ) ) != 0 )
+                                output, &enclen ) ) != 0 )
     {
         goto exit;
     }
+    *olen += enclen;
 
-    if( ( ret = mbedtls_cipher_finish( &cipher_ctx, output + olen, &olen ) ) != 0 )
+    if( ( ret = mbedtls_cipher_finish( &cipher_ctx, output + enclen, &enclen ) ) != 0 )
         ret = MBEDTLS_ERR_PKCS12_PASSWORD_MISMATCH;
+    *olen += enclen;
 
 exit:
     mbedtls_platform_zeroize( key, sizeof( key ) );
```

Comment:
Do you think we should zeroize the output in case of failure?

Reply:
Good question. I dont really think it matters much but we could do it if we want to be a bit paranoid. What do you think?

Reply:
I think that in case of partial decryption, it would be good to clear what was already decrypted, to avoid cache attacks